app/lib/timetable/timetable_page/school_class_filter/school_class_filter.dart

```diff
@@ -96,11 +100,29 @@ class _SchoolClassFilterBottomBarState
   Future<void> _openAndHandleSelectionMenu(BuildContext context,
       TimetableBloc bloc, SchoolClassFilterView view) async {
     final selectedOption = await _openSelectionMenu(view);
-    if (selectedOption != null) {
+    if (selectedOption == null) {
+      return;
+    }
+
+    if (_shouldOpenSharezonePlusPage(context, selectedOption)) {
+      _openSharezonePlusPage(context);
+    } else {
       bloc.changeSchoolClassFilter(selectedOption);
     }
   }
 
+  bool _shouldOpenSharezonePlusPage(
+      BuildContext context, SchoolClassFilter selectedOption) {
+    final isPlusEnabled = context.read<SubscriptionEnabledFlag>().isEnabled;
+    final hasPlus = context.read<SubscriptionService>().isSubscriptionActive();
+    return selectedOption.isPlusRequiredFilter && !hasPlus && isPlusEnabled;
+  }
```

Comment:
```suggestion
  bool _shouldOpenSharezonePlusPage(
      BuildContext context, SchoolClassFilter selectedOption) {
    final isUnlocked = context
        .read<SubscriptionService>()
        .hasFeatureUnlocked(SharezonePlusFeature.filterTimetableByClass);
    return selectedOption.shouldFilterForClass && !isUnlocked;
  }
```

We already implemented `SubscriptionService` in such a way, that if Sharezone Plus is not activated then all features are unlocked.

I think for the change above you would need to create `SharezonePlusFeature.filterTimetableByClass` and maybe even a new `SubscriptionTier`?

Also I removed the `SchoolClassFilter.isPlusRequiredFilter` which I mean is not wrong, but I felt a bit uneasy with this (I'm not sure if the logic should be on the `SchoolClassFilter`). This is just a hunch though, you are free to keep it there.

Reply:
Ah, I forgot about `hasFeatureUnlocked` 😂